### DISCUSSION TITLE
Koyamin您好，请问封面左上角的2077应该怎么改成2076

### DISCUSSION THREAD

**Discussion by zhanhonhao** [Koyamin/ecnuthesis#12] [Q&A] [answered]
（2023年完成论文，但是2022届本科生），我尝试看懂.cls文件但是发现我根本不会改，orz感谢orz

**Comment by SilenceMonk** [comment]
> （2023年完成论文，但是2022届本科生），我尝试看懂.cls文件但是发现我根本不会改，orz感谢orz

同问！我在.cls文件里找到了这个

https://github.com/Koyamin/ECNUThesis-Undergraduate/blob/d3f937b20e969ae13a15d202643ce920f06c772d/ecnuthesis.cls#L884

是不是能够找到办法算出```\year-1```就能解决？

**Comment by CCXXXI** [comment]
`\theyear` 直接换成 `2022` 就行

**Comment by SilenceMonk** [comment]
家人们搞定了

这里定义了一个变量
https://github.com/Koyamin/ECNUThesis-Undergraduate/blob/d3f937b20e969ae13a15d202643ce920f06c772d/ecnuthesis.cls#L154

这里定义了enrollYear命令
https://github.com/Koyamin/ECNUThesis-Undergraduate/blob/d3f937b20e969ae13a15d202643ce920f06c772d/ecnuthesis.cls#L185-L189

可以把enrollYear里边的```-4```改成```-1```，然后把下面的```\theyear```改成```\enrollYear```
https://github.com/Koyamin/ECNUThesis-Undergraduate/blob/d3f937b20e969ae13a15d202643ce920f06c772d/ecnuthesis.cls#L884

应该就可以了，或者仿照这个enrollYear新写一个命令也行

**Comment by SilenceMonk** [comment]
> `\theyear` 直接换成 `2022` 就行

哦哦好滴！

**Comment by Koyamin** [comment]
https://github.com/Koyamin/ECNUThesis-Undergraduate/commit/cec5185fda3982fd6d12ab68fd1404834f76482c

在 `info` 选项中增加了一个接口 `graduationYear`，用于指定内封面毕业届别。若该接口未定义，则内封面毕业届别将被默认设置为 `year` 的值。

**Comment by SilenceMonk** [comment]
> [cec5185](https://github.com/Koyamin/ECNUThesis-Undergraduate/commit/cec5185fda3982fd6d12ab68fd1404834f76482c)
> 
> 在 `info` 选项中增加了一个接口 `graduateYear`，用于指定内封面毕业届别。若该接口未定义，则内封面毕业届别将被默认设置为 `year` 的值。

感谢Koyamin大大！

**Comment by Koyamin** [comment, marked as answer]
> （2023年完成论文，但是2022届本科生），我尝试看懂.cls文件但是发现我根本不会改，orz感谢orz

按照教务处要求，存档板毕业论文的届数与写作时间均应相同。
